logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt

```diff
@@ -71,8 +71,9 @@ class UserDataSource(
                 Either.Left(it)
             }, {
                 val usersToBePersisted = it.map(userMapper::fromApiModelToDaoModel)
-                // Save (in case there is no data) a reference to the asset id (profile picture)
-                assetRepository.saveUserPictureAsset(mapAssetsForUsersToBePersisted(usersToBePersisted))
+                // not sure about this, because on first sync it's taking ages =/
+                // options, doing it in 2 steps like before and downloading on demand or doing it somehow in a non.blocking fashion
+                assetRepository.downloadUsersPictureAssets(mapAssetsForUsersToBePersisted(usersToBePersisted))
```

Comment:
Yes, we shouldn't do this here but on demand.

Reply:
You already have everything for this though?

GetPublicAssetUseCase(user. previewPicture)

Reply:
I see an issue with this tough, because we have a `FOREIGN` KEY constraint on `Users` Table pointing to the `Assets` table. So the first time we need to persist in the assets table first, then the user (this was the reason of having the empty asset insert 😓 )

The option to achieve the on demand execution, that I also agree, will be removing the constraint and have the relation only at the Kotlin Model level 🤔  Would that make sense ?

Reply:
Applied the changes, now we are not using anymore the FK constraint, and we are able to do the sync on demand later via usecase